### PR TITLE
Add support for suitable flag to data local service

### PR DESCRIPTION
**Pull request by pschorf**
## Changes proposed in this PR
- Adds support for a "suitable" flag on data local service responses

## Why are we making these changes?
This allows more granular control of the scheduling by the data local service.

### CODE REVIEW

File: scheduler/src/cook/mesos/constraints.clj [hunk 1]
```diff
@@ -136,18 +136,22 @@
...
           launch-after-age (t/plus (tc/from-date submit-time) (t/seconds launch-wait-seconds))
           launch-without-data? (t/after? (t/now) launch-after-age)]
-      (if (or launch-without-data?
-              (contains? data-locality-costs datasets))
-        [true nil]
-        [false "No data locality costs available"])))
-  (job-constraint-evaluate [this _ _ _]
-    (job-constraint-evaluate this _ _)))
+      (if (contains? data-locality-costs datasets)
+        (let [{:keys [suitable]} (get-in data-locality-costs [datasets (get vm-attributes "HOSTNAME")] {:suitable true})]
```
Comment (shamsimam) [inline scheduler/src/cook/mesos/constraints.clj]:
```suggestion
        (let [{:keys [suitable] :or {suitable true}} (get-in data-locality-costs [datasets (get vm-attributes "HOSTNAME")])]
```

File: scheduler/src/cook/mesos/data_locality.clj [hunk 2]
```diff
@@ -198,8 +200,10 @@
           {:keys [job/uuid] :as job} (:job task-request)
           datasets (get-dataset-maps job)]
       (if-not (empty? (or datasets []))
-        (let [normalized-fitness (- 1.0
-                                    (get-in (get-data-local-costs) [datasets (.getHostname target-vm)] 1.0))
+        (let [{:keys [cost suitable]}(get-in (get-data-local-costs) [datasets (.getHostname target-vm)] {:cost 1.0
```
Comment (shamsimam) [inline scheduler/src/cook/mesos/data_locality.clj]: Nitpick: missing space.

Comment (shamsimam) [reply scheduler/src/cook/mesos/data_locality.clj]: Can we use `:or` in the destructuring to assign the default values for `cost` and `suitable`?

File: scheduler/src/cook/mesos/constraints.clj (hunk 1 above)
Comment (pschorf) [reply scheduler/src/cook/mesos/constraints.clj]: Done

File: scheduler/src/cook/mesos/data_locality.clj (hunk 2 above)
Comment (pschorf) [reply scheduler/src/cook/mesos/data_locality.clj]: Done

**Pull request merged**